laws/shared/src/test/scala/algebra/laws/LawTests.scala

```diff
@@ -44,13 +44,22 @@ trait LawTestsBase extends FunSuite with Discipline {
   laws[LogicLaws, Boolean].check(_.bool)
   laws[LogicLaws, SimpleHeyting].check(_.heyting)
   laws[LatticePartialOrderLaws, Boolean].check(_.boundedLatticePartialOrder)
-  laws[RingLaws, Boolean].check(_.boolRing(booleanRing))
+  laws[RingLaws, Boolean].check { l =>
+    implicit val br: BoolRing[Boolean] = booleanRing
```

Comment:
Yeah, this is the biggest bummer to me, because the other changes aren't particularly pretty, but they are at least hidden within the laws. This affects the user code that wants to check laws.

I'm curious if it would be any cleaner to use the "reader" approach and represent laws as `checkSerializability => A`. The `check` method could be a helper method for when you want serializability to be checked, and there could be an alternative `checkWithoutSerializability` or something that fixes the input to not check serializability. This might help with the changes in this file, since there would no longer be two implicit parameters. It would also help with the places like https://github.com/non/algebra/pull/129/files#r47260083. I'm unsure how clean it would look in general though.